### DISCUSSION TITLE
Loading LCM-Lora after or before loading ip-adapter?

### DISCUSSION THREAD

**Discussion by juancopi81** [huggingface/diffusers#6642] [General]
Hi,

I was looking at the docs and found that it is better to load the ip-adapter weights before loading the LCM-lora:

"_Note that you need to load IP-Adapter weights before loading the LCM-Lora weights_."

```
pipe.load_ip_adapter("h94/IP-Adapter", subfolder="models", weight_name="ip-adapter_sd15.bin")
pipe.load_lora_weights(lcm_lora_id)
```

For my use case, it is better to have loaded the LCM-lora and at some point load the ip-adapter. I tried both loading after and before the ip-adapter and the results are the same (at least visually). 

Is this expected?

**Comment by sayakpaul** [comment]
Thanks for initiating. 

Which order of loading gives you the expected results?

**Comment by juancopi81** [comment]
Hi @sayakpaul and thank you very much for your answer. Loading the ip-adapter before and after loading the LCM-Lora seems to have no effect (they produce the same image). I was wondering if there is any reason why the docs recommend first loading the ip-adapter and the the LCM-Lora. 

I found [here](https://huggingface.co/docs/diffusers/main/en/using-diffusers/loading_adapters#lcm-lora):

_"Note that you need to load IP-Adapter weights before loading the LCM-Lora weights."_

I tested both scenarios following this code:

```py
from diffusers import DiffusionPipeline, LCMScheduler
import torch
from diffusers.utils import load_image

model_id =  "sd-dreambooth-library/herge-style"
lcm_lora_id = "latent-consistency/lcm-lora-sdv1-5"

pipe = DiffusionPipeline.from_pretrained(model_id)

# For the first test
pipe.load_ip_adapter("h94/IP-Adapter", subfolder="models", weight_name="ip-adapter_sd15.bin")
pipe.load_lora_weights(lcm_lora_id)
pipe.scheduler = LCMScheduler.from_config(pipe.scheduler.config)
# For the second test
# pipe.load_ip_adapter("h94/IP-Adapter", subfolder="models", weight_name="ip-adapter_sd15.bin")

prompt = "best quality, high quality"
image = load_image("https://user-images.githubusercontent.com/24734142/266492875-2d50d223-8475-44f0-a7c6-08b51cb53572.png")

# Seed for reproductibility
generator = torch.Generator(device="cpu").manual_seed(32)
images = pipe(
    prompt=prompt,
    ip_adapter_image=image,
    num_inference_steps=4,
    guidance_scale=1,
    generator=generator
).images[0]
```

And both produce the same image:
Image in first scenario (loading the ip-adapter first):
![first_ip_adapter](https://github.com/huggingface/diffusers/assets/4313860/def53b76-4acf-4ea5-adf4-952c883be82a)

Image in second scenario (loading the LCM_Lora first)
![first_lcm_lora](https://github.com/huggingface/diffusers/assets/4313860/79ed4a41-fd97-42e3-8c2b-95b6f255bb8c)

So I was wondering if there is any reason the docs recommend one approach over the other.

Thanks again!!

**Reply by sayakpaul** [reply]
I will let @patil-suraj answer further. Cc @yiyixuxu too for awareness.

**Comment by yiyixuxu** [comment]
We can maybe remove that line from the doc?

**Reply by juancopi81** [reply]
Thanks @yiyixuxu  I did no notice any difference, but just wanted to ask if I am missing something :)

**Reply by skywalker00001** [reply]
There should be no difference. You can verify by substracting 1img from 2img, the result is supposed to be 0.
